### PR TITLE
JBPM-9659: Upgrade deployments to WildFly 23

### PR DESCRIPTION
**Pull request by jesuino**
**JIRA**: 

[JBPM-9659](https://issues.redhat.com/browse/JBPM-9659)

**Related PRs*: 

https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1693
https://github.com/kiegroup/kie-soup/pull/216
https://github.com/kiegroup/droolsjbpm-knowledge/pull/543
https://github.com/kiegroup/drools/pull/3757
https://github.com/kiegroup/optaplanner/pull/1463
https://github.com/kiegroup/lienzo-tests/pull/124
https://github.com/kiegroup/appformer/pull/1163
https://github.com/kiegroup/kie-uberfire-extensions/pull/122
https://github.com/kiegroup/jbpm/pull/1974
https://github.com/kiegroup/kie-jpmml-integration/pull/62
https://github.com/kiegroup/droolsjbpm-integration/pull/2539
https://github.com/kiegroup/kie-wb-playground/pull/115
https://github.com/kiegroup/kie-wb-common/pull/3660
https://github.com/kiegroup/drools-wb/pull/1506
https://github.com/kiegroup/jbpm-designer/pull/913
https://github.com/kiegroup/jbpm-work-items/pull/212
https://github.com/kiegroup/jbpm-wb/pull/1506
https://github.com/kiegroup/optaplanner-wb/pull/399
https://github.com/kiegroup/kie-wb-distributions/pull/1115
https://github.com/kiegroup/openshift-drools-hacep/pull/116
https://github.com/kiegroup/process-migration-service/pull/5
https://github.com/kiegroup/optaweb-employee-rostering/pull/662
https://github.com/kiegroup/optaweb-vehicle-routing/pull/535

### CODE REVIEW

File: repository/pom.xml [hunk 1]
```diff
@@ -504,6 +493,12 @@
           <artifactId>jira-workitem</artifactId>
           <type>zip</type>
           <scope>provided</scope>
+          <exclusions>
+            <exclusion>
+              <groupId>org.jvnet</groupId>
+              <artifactId>mimepull</artifactId>
+            </exclusion>
+          </exclusions>
```
Comment (tiagobento) [inline repository/pom.xml:501]: Not sure about this one.

**Comment by tiagobento** [comment]
jenkins run fdb

File: repository/pom.xml (hunk 1 above)
Comment (elguardian) [reply repository/pom.xml:506]: if the scope is provided it should not be a big deal as long as jira-workitem should have that dependency as compiled already.

**Comment by elguardian** [comment]
failure about kafka reconnect should not be too much of a problem.

**Review by elguardian — APPROVED** [review]

**Comment by elguardian** [comment]
Jenkins retest this please.

**Review by mareknovotny — APPROVED** [review]

**Comment by mareknovotny** [comment]
`org.jbpm.workitem.springboot.samples.KafkaProxySampleTest.testKafkaWIHReconnect` looks like a connection env problem? I saw this multiple times for that test.

jenkins retest this please

**Review by tomasdavidorg — APPROVED** [review]

File: repository/pom.xml (hunk 1 above)
Comment (tiagobento) [reply repository/pom.xml]: Removed in 3c1e362

**Comment by mareknovotny** [comment]
hmm, the last removal of exclusion seems show the exclusion is needed at least on GHA jdk11 build?
Notice that Jenkins is on jdk 8 and GHA jdk 8 also is fine but fails on Kafka tests.

`2021-08-17T21:01:12.9440481Z [WARNING] Rule 0: org.apache.maven.plugins.enforcer.BanDuplicateClasses failed with message:
2021-08-17T21:01:12.9442113Z Duplicate classes found:
2021-08-17T21:01:12.9442897Z 
2021-08-17T21:01:12.9443385Z   Found in:
2021-08-17T21:01:12.9444076Z     org.jvnet.mimepull:mimepull:jar:1.9.8:provided
2021-08-17T21:01:12.9444833Z     org.jvnet:mimepull:jar:1.4:provided
2021-08-17T21:01:12.9445477Z   Duplicate classes:
2021-08-17T21:01:12.9446378Z     org/jvnet/mimepull/FileData.class
2021-08-17T21:01:12.9447162Z     org/jvnet/mimepull/FinalArrayList.class
2021-08-17T21:01:12.9448011Z     org/jvnet/mimepull/InternetHeaders.class
2021-08-17T21:01:12.9448798Z     org/jvnet/mimepull/MIMEParser$1.class
2021-08-17T21:01:12.9449503Z     org/jvnet/mimepull/MIMEPart.class
2021-08-17T21:01:12.9450226Z     org/jvnet/mimepull/MIMEMessage$1.class
2021-08-17T21:01:12.9451066Z     org/jvnet/mimepull/DataHead$ReadMultiStream.class
2021-08-17T21:01:12.9451916Z     org/jvnet/mimepull/WeakDataFile.class
2021-08-17T21:01:12.9452735Z     org/jvnet/mimepull/MIMEEvent$EndMessage.class
2021-08-17T21:01:12.9453511Z     org/jvnet/mimepull/DataFile.class
2021-08-17T21:01:12.9454309Z     org/jvnet/mimepull/MIMEConfig.class
2021-08-17T21:01:12.9455436Z     org/jvnet/mimepull/MIMEParser$MIMEEventIterator.class
2021-08-17T21:01:12.9456463Z     org/jvnet/mimepull/MIMEParser$LineInputStream.class
2021-08-17T21:01:12.9457176Z     org/jvnet/mimepull/MIMEEvent$Headers.class
2021-08-17T21:01:12.9457777Z     org/jvnet/mimepull/Chunk.class
2021-08-17T21:01:12.9458337Z     org/jvnet/mimepull/MemoryData.class
2021-08-17T21:01:12.9458942Z     org/jvnet/mimepull/MIMEEvent$EVENT_TYPE.class
2021-08-17T21:01:12.9459496Z     org/jvnet/mimepull/Header.class
2021-08-17T21:01:12.9460125Z     org/jvnet/mimepull/DataHead$ReadOnceStream.class
2021-08-17T21:01:12.9460804Z     org/jvnet/mimepull/MIMEEvent$Content.class
2021-08-17T21:01:12.9461378Z     org/jvnet/mimepull/MIMEEvent.class
2021-08-17T21:01:12.9461944Z     org/jvnet/mimepull/MIMEParser.class
2021-08-17T21:01:12.9462644Z     org/jvnet/mimepull/MIMEParsingException.class
2021-08-17T21:01:12.9463384Z     org/jvnet/mimepull/MIMEEvent$StartMessage.class
2021-08-17T21:01:12.9464091Z     org/jvnet/mimepull/ChunkInputStream.class
2021-08-17T21:01:12.9464703Z     org/jvnet/mimepull/DataHead.class
2021-08-17T21:01:12.9465275Z     org/jvnet/mimepull/MIMEMessage.class
2021-08-17T21:01:12.9465815Z     org/jvnet/mimepull/Data.class
2021-08-17T21:01:12.9466349Z     org/jvnet/mimepull/MIMEEvent$EndPart.class
2021-08-17T21:01:12.9467133Z     org/jvnet/mimepull/MIMEParser$STATE.class
2021-08-17T21:01:12.9467768Z     org/jvnet/mimepull/MIMEEvent$StartPart.class
`

**Comment by tiagobento** [comment]
@mareknovotny @elguardian Yeah.. so it's okay if I put the exclusion back?

**Comment by mbiarnes** [comment]
Build was successful - only one test failed 
https://eng-jenkins-csb-business-automation.apps.ocp4.prod.psi.redhat.com/job/KIE/job/main/job/pullrequest/job/jbpm-work-items-main.pr/19/testReport/junit/org.jbpm.workitem.springboot.samples/KafkaProxyAsyncSampleTest/testAsyncKafkaWIHNoConnection_0__SINGLETON_/

**Pull request merged**